build failure due to calling cudaThreadSynchronize()

I'm currently getting a build error of:

```
/home/wald/Projects/OpenSubdiv/opensubdiv/osd/cudaEvaluator.cpp:362:5: error: ‘cudaThreadSynchronize’ was not declared in this scope; did you mean ‘cudaStreamSynchronize’?
  362 |     cudaThreadSynchronize();
      |     ^~~~~~~~~~~~~~~~~~~~~

```
I assume that is a left-over from (much) older cuda versions that can be safely removed, and am happy to send a PR that removes this - but wanted to double-check that this isn't something else that I just don't understand. Safe to remove?

--- a/opensubdiv/osd/cudaEvaluator.cpp
+++ b/opensubdiv/osd/cudaEvaluator.cpp
@@ -359,7 +359,7 @@ CudaEvaluator::EvalPatches(
 /* static */
 void
 CudaEvaluator::Synchronize(void * /*deviceContext*/) {
-    cudaThreadSynchronize();
+    cudaDeviceSynchronize();
 }
 
 }  // end namespace Osd

